mico-core/src/main/java/io/github/ust/mico/core/broker/MicoApplicationBroker.java

```diff
@@ -249,17 +294,86 @@ public MicoApplication removeMicoServiceFromMicoApplicationByShortNameAndVersion
             // Application does not include the service -> cannot not be deleted from it
             throw new MicoApplicationDoesNotIncludeMicoServiceException(applicationShortName, applicationVersion, serviceShortName);
         } else {
-            // 1. Remove the service from the application
-            micoApplication.getServices().removeIf(s -> s.getShortName().equals(serviceShortName));
-            MicoApplication updatedMicoApplication = applicationRepository.save(micoApplication);
-            // 2. Delete the corresponding service deployment information
+            // 1. Delete the corresponding service deployment information
             serviceDeploymentInfoRepository.deleteByApplicationAndService(applicationShortName, applicationVersion, serviceShortName);
-            return updatedMicoApplication;
+            // 2. Remove the service from the application
+            micoApplication.getServices().removeIf(s -> s.getShortName().equals(serviceShortName));
+            return applicationRepository.save(micoApplication);
         }
 
         // TODO: Update Kubernetes deployment (see issue mico#627)
     }
 
+    public KFConnectorDeploymentInfo addKafkaFaasConnectorInstanceToMicoApplicationByVersion(
```

Comment:
Missing javadoc

Reply:
Currently no public method in this class has a JavaDoc. If a JavaDoc is important here, we should do add them in a separate issue!?

Reply:
While that may be true every added or refactored public method should have javadoc. Omitting it because the rest of the file also does not have it is not a good reason.